frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { useParams } from 'react-router';
+
+import useGetCourseInfo from 'api/courses/getCourseInfo';
+import useApplyCourse from 'api/courses/applyCourse';
+
+import DetailedCourse from './DetailedCourse';
+
+const DetailedCourseContainer: React.FC = () => {
+  const params = useParams();
+  const { data } = useGetCourseInfo(params._id);
+  const mutation = useApplyCourse();
```

Comment:
We should de from the start what the props here:
```suggestion
  const { mutate, isLoading: isLoadingApplying } = useApplyCourse();
```